### PR TITLE
q.33 Addition

### PR DESCRIPTION
**Pull request by wbaik**
Q.33 added.
I love one liner fib, so adding it as well... I learned an even faster fib O(log N), but this is more than enough.

### CODE REVIEW

File: src/mp/math_utils.py [hunk 1]
```diff
@@ -41,4 +42,4 @@ def get_distinct_prime_facs(num):
 
 # One-liner fibonacci for a record
 def fib(n):
-    return reduce(lambda item, _: (item[1], item[0] + item[1]), range(n-1), (1, 1))[0]
+    return reduce(lambda item, _: (item[1], item[0] + item[1]), range(n), (0, 1))[1]
```
Comment (virgax7) [inline src/mp/math_utils.py]: Beautiful functional fibonacci.
I think you meant to leave the [0] as [0] though so
`return reduce(lambda item, _: (item[1], item[0] + item[1]), range(n), (0, 1))[0]`

File: src/questions-21-40/q33.py [hunk 2]
```diff
@@ -7,5 +7,29 @@
...
+    nums, denums = zip(*temps)
+    return nums, denums
+
+
+nums, denums = get_curious_fractions()
+
+numerators = functools.reduce(mul, nums)
+denomenators = functools.reduce(mul, denums)
+
+print(denomenators/math.gcd(numerators, denomenators))
```
Comment (virgax7) [inline src/questions-21-40/q33.py:31]: :D nice code mate! I didn't know this was what it wanted

File: src/mp/math_utils.py (hunk 1 above)
Comment (wbaik) [reply src/mp/math_utils.py]: Yea, people seem to use different conventions, so if it starts with fib(0) = 1 or 0 makes difference sometimes...

File: src/questions-21-40/q33.py (hunk 2 above)
Comment (wbaik) [reply src/questions-21-40/q33.py:31]: Yea, I figure it was really this weird sentences getting you confused. Seem like an easy question relative to others you're doing these days.

**Review by wbaik — COMMENTED** [review]
Fib convention

File: src/questions-21-40/q33.py (hunk 2 above)
Comment (virgax7) [reply src/questions-21-40/q33.py:31]: maybe for your mathematical mind haha

**Comment by virgax7** [comment]
Excellent

**Comment by wbaik** [comment]
Thanks boss

**Pull request merged**